pkg/resource/graph/dependency_graph.go

```diff
@@ -288,6 +288,18 @@ func (dg *DependencyGraph) ChildrenOf(res *resource.State) []*resource.State {
 	return children
 }
 
+// ParentsOf returns a slice containing all resources that are parents of the given resource.
+func (dg *DependencyGraph) ParentsOf(res *resource.State) []*resource.State {
+	parents := make([]*resource.State, 0)
+	for i := len(dg.resources) - 1; i >= 0; i-- {
+		if dg.resources[i].URN == res.Parent {
+			parents = append(parents, dg.resources[i])
+			res = dg.resources[i]
+		}
+	}
+	return parents
+}
```

Comment:
So how I think this works:

* We're expecting resources to be topologically sorted, so by walking backwards we know that whenever we match a parent, we have yet to see _that_ parent's parent (if it exists)
* We know it's safe to terminate when we've traversed the full set in reverse.

If that's right, thoughts on a comment to that effect? It seems we have other bits in this file where we explain the logic of reverse traversal etc. (Side-note: it's a shame that we don't document that `dg.resources` basically _must_ be topologically sorted for any of these functions to work, outside of where we create a DG).

Also, I think a test or two for `ParentsOf` could be valuable.

Reply:
Yup, that's right.  I've added a comment, and a couple of tests for `ParentsOf`.